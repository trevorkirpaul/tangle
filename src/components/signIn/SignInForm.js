import React from 'react';

export default () => {
  return (
    <form>
      <input type="text" name="email" placeholder="email" />
      <input type="password" name="password" placeholder="password" />

      <button type="submit">Sign In</button>
    </form>
  );
};
