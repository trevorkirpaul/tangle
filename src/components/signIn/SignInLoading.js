import React from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import Progress from 'material-ui/CircularProgress';

export default ({
  open,
  disabled,
  failed,
  handleContinue,
  handleClose,
  statusMessage,
  loading,
}) => {
  const actions = [
    <FlatButton
      label="continue"
      disabled={disabled}
      primary={true}
      onClick={handleContinue}
    />,
  ];
  const failedActions = [
    <FlatButton label="okay" primary={true} onClick={handleClose} />,
  ];
  return (
    <Dialog
      title={statusMessage}
      actions={failed ? failedActions : actions}
      modal={true}
      open={open}
    >
      {loading ? <Progress /> : <p>{statusMessage}</p>}
    </Dialog>
  );
};
