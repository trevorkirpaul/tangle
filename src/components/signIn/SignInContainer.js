import React, { Component } from 'react';
import SignInForm from './SignInForm';
import { connect } from 'react-redux';
import { signIn } from '../../actions/auth';

export class SignInContainer extends Component {
  render() {
    return (
      <div>
        <SignInForm />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  signIn: fields => dispatch(signIn(fields)),
});

export default connect(null, mapDispatchToProps)(SignInContainer);
