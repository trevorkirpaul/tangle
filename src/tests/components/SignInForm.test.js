import React from 'react';
import { shallow } from 'enzyme';
import { SignInForm } from '../../components/signIn/SignInForm';
import { capFirstLetter } from '../../components/signIn/SignInForm';

test('should render container correctly', () => {
  const wrapper = shallow(<SignInForm />);
  expect(wrapper).toMatchSnapshot();
});

// helper function in SignInForm
test('should capitalize first letter in a given string', () => {
  const string = 'word';
  expect(capFirstLetter(string)).toEqual('Word');
});
