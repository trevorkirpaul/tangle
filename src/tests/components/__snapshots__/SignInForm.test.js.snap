// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render container correctly 1`] = `
<styled.form>
  <styled.div>
    <TextField
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      floatingLabelText="email"
      fullWidth={false}
      multiLine={false}
      name="email"
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
    />
  </styled.div>
  <styled.div>
    <TextField
      disabled={false}
      errorText=""
      floatingLabelFixed={false}
      floatingLabelText="password"
      fullWidth={false}
      multiLine={false}
      name="password"
      onChange={[Function]}
      rows={1}
      type="password"
      underlineShow={true}
    />
  </styled.div>
  <RaisedButton
    disabled={false}
    fullWidth={false}
    label="Sign In"
    labelPosition="after"
    onClick={[Function]}
    primary={true}
    secondary={false}
  />
</styled.form>
`;
